From fbe55b3493876cf6ef823c90f312c4bedac3eeeb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Zbigniew=20J=C4=99drzejewski-Szmek?= <zbyszek@in.waw.pl>
Date: Tue, 24 Apr 2018 13:46:58 +0200
Subject: [PATCH] basic/audit-util: always log the reason when disabling audit
 logs

This state is cached, and it's seems OK to log at least once.

(cherry picked from commit 13bb68bbe37f0b39cd45234b09fb1a1da8302020)
---
 src/basic/audit-util.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/src/basic/audit-util.c b/src/basic/audit-util.c
index 6a93c9109c..1e897cb982 100644
--- a/src/basic/audit-util.c
+++ b/src/basic/audit-util.c
@@ -95,10 +95,9 @@ bool use_audit(void) {
                 fd = socket(AF_NETLINK, SOCK_RAW|SOCK_CLOEXEC|SOCK_NONBLOCK, NETLINK_AUDIT);
                 if (fd < 0) {
                         cached_use = !IN_SET(errno, EAFNOSUPPORT, EPROTONOSUPPORT, EPERM);
-                        if (errno == EPERM)
-                                log_debug_errno(errno, "Audit access prohibited, won't talk to audit");
-                }
-                else {
+                        if (!cached_use)
+                                log_debug_errno(errno, "Won't talk to audit: %m");
+                } else {
                         cached_use = true;
                         safe_close(fd);
                 }
-- 
2.17.0

